core/templates/dev/head/profile/PreferencesSpec.js

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Preferences Controller', function() {
+  describe('PreferencesCtrl', function() {
+    var scope, ctrl, $Http;
+
+    beforeEach(function() {
+      module('oppia');
+    });
+
+    beforeEach(inject(function(_$httpBackend_, $http, $rootScope, $controller) {
+      _$httpBackend_.expectGET('/signuphandler/data').respond({
```

Comment:
So -- I have absolutely no idea what's going on in this test suite. What you probably want here is an expectGET for the /preferenceshandler/data URL; that will tell the test suite: "when the test makes a GET request to the URL -- and it will do so! -- then give it this thing back". I don't think the preferences page makes a call to /signuphandler/data.

Note that, in the frontend karma tests, there is no backend server running at all. Everything is happening only in the frontend. If you want to test persistence, a protractor e2e test would be more appropriate -- in such a test, both the client and the server are running.

I'll make some more comments on this file, but they are mainly just FYI about how karma tests work. On the whole, you'd probably want to replace this whole thing with a single protractor test.